packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,333 @@
+        html<TableColumnMapping, TableColumnMappingCellView>`
+            ${when((x, c) => x.key === (c.parent as TableColumnMappingCellView).cellRecord.value,
+                html<TableColumnMapping>`
+                    ${repeat(x => x.mappedTemplate,
+                        html<HTMLTemplateElement>`${x => new ViewTemplate(x.innerHTML, [])}`
+                    )}
+                `
+            )}
+        `
+    )}
+</div>
+```
+
+Header view:
+
+Note the following requires that `TableColumnMappingHeaderView` has a reference to the column with which it is associated. This is needed to enumerate the column's mapping elements.
+
+```HTML
+<div
+    ${ref('wrapper')}
+    class="wrapper"
+    @mouseover="${x => {
+        x.isValidContentAndHasOverflow = !!x.wrapper.innerText && x.wrapper.offsetWidth < x.wrapper.scrollWidth;
+    }}"
+    @mouseout="${x => {
+        x.isValidContentAndHasOverflow = false;
+    }}"
+    title=${x => (x.isValidContentAndHasOverflow ? x.wrapper.innerText : null)}
+>
+    ${repeat(x => (x.column as TableColumnMapping).mappings,
+        html<TableColumnMapping, TableColumnMappingHeaderView>`
+            ${when((x, c) => x.key === (c.parent as TableColumnMappingHeaderView).groupHeaderValue,
+                html<TableColumnMapping>`
+                    ${repeat(x => x.mappedTemplate,
+                        html<HTMLTemplateElement>`${x => new ViewTemplate(x.innerHTML, [])}`
+                    )}
+                `
+            )}
+        `
+    )}
+</div>
+```
+
+For each mapping, if the key matches the value for this row, get the mapping's contained `template` and render its `innerHTML`.
+
+#### `nimble-mapping-string/number/boolean`
+
+```HTML
+<template slot="mapping">
+    <slot
+        ${slotted({ property: 'mappedTemplate', filter: x => x instanceof HTMLTemplateElement })}
+    >
+    </slot>
+</template>
+```
+
+### Grouping
+
+Will support grouping by the record value. The grouping header will display the rendered html mapped to that value. If multiple different values map to the same rendered html (e.g. icon), there may be separate groups that visually look like they should be combined into one group. We could instead render the record value into the header, but that value is not otherwise exposed to the user. It may not even have much meaning (e.g. some numeric code). I think it is more reasonable to display the rendered html.
+
+### Sorting
+
+Sorting has the same issue as grouping, where visually identical cells could end up not sorted contiguously. Again, I believe this is reasonable.
+
+### Sizing
+
+We will support fixed or fractional widths for this column type. We will introduce a new mixin for fixed-width support that exposes a `pixel-width` property. If this is set, the column will have a fixed width, otherwise it defaults to fractional width. There is an undocumented minimum column width, so a client may set `pixel-width` to 1 to end up with that minimum width.
```

Comment:
I'm not sure I understand the intent here. When you say "we will support fixed _or_ fractional widths", are you suggesting that the user be able to configure the fractional width setting (I would expect this)? If so, then there should not be an "undocumented" minimum column width, that comes with the `FractionalWidthColumn` mixin. Additionally, this would suggest that the API (described above) should include `fractional-width` and `minimum-pixel-width`.

If the intent is to _not_ allow fractional width adjustment, then I think we should have it behave in such a way that it _never_ behaves in such a way. This would require us to provide a default for `pixel-width` that is non-null/undefined, and also handle the case in the mixin where a user sets it to null/undefined and have it reset to the provided default (since a pixel-width column must have a non-null/undefined value in order to not be sized fractionally).

Reply:
Updated the spec to describe new sizing plan.